plugins/libraries.ts

```diff
@@ -0,0 +1,66 @@
+import { Plugin } from '@nuxt/types/app';
+
+type LibrariesType = {
+  fetchItem: (id: string) => Promise<void>;
+  fetchItems: (params: Record<string, unknown>) => Promise<string[]>;
+};
+
+declare module '@nuxt/types' {
+  interface Context {
+    $libraries: LibrariesType;
```

Comment:
This could probably use a different name since this one is a bit confusing (It implies this plugin deals with libraries, but it doesn't.... it deals with fetching items).

Reply:
Yup I wanted to think a little bit more about it at a later stage, but feel free to say it if you already got one ^^

Reply:
Though a bit more about this. This plugin should probably be a fairly simple wrapper around the userLibrary API itself.

So naming it `$userLibrary` would make sense.

Furthermore, it should expose the same functions as the API itself, with the same names and parameters, but essentially handle the items store for us automatically.